src/custom/state/enhancedTransactions/updater/FinalizeTxUpdater.tsx

```diff
@@ -137,25 +143,36 @@ function finalizeEthFlowTx(
   }
 
   if (subType === 'cancellation') {
-    if (receipt.status === 1) {
-      // If cancellation succeeded, mark order as cancelled
-      dispatch(cancelOrdersBatch({ chainId, ids: [orderId] }))
-    } else {
-      // If cancellation failed:
-      // 1. Update order state and remove the isCancelling flag and cancellationHash
-      dispatch(updateOrder({ chainId, order: { id: orderId, isCancelling: false, cancellationHash: undefined } }))
-      // 2. Show failure tx pop-up
-      addPopup(
-        {
-          txn: {
-            hash,
-            success: false,
-            summary: `Failed to cancel order selling ${nativeCurrencySymbol}`,
-          },
+    finalizeOnChainCancellation(receipt, params, hash, orderId)
+  }
+}
+
+function finalizeOnChainCancellation(
+  receipt: TransactionReceipt,
+  params: CheckEthereumTransactions,
+  hash: string,
+  orderId: string
+) {
+  const { chainId, dispatch, addPopup, nativeCurrencySymbol } = params
+
+  if (receipt.status === 1) {
+    // If cancellation succeeded, mark order as cancelled
+    dispatch(cancelOrdersBatch({ chainId, ids: [orderId] }))
+  } else {
+    // If cancellation failed:
+    // 1. Update order state and remove the isCancelling flag and cancellationHash
+    dispatch(updateOrder({ chainId, order: { id: orderId, isCancelling: false, cancellationHash: undefined } }))
+    // 2. Show failure tx pop-up
+    addPopup(
+      {
+        txn: {
+          hash,
+          success: false,
+          summary: `Failed to cancel order selling ${nativeCurrencySymbol}`,
```

Comment:
Does `nativeCurrencySymbol` also apply for non ethFlow cancellations?

Reply:
Nice catch! Thank you!

Reply:
Fixed